lib/widgets.php

```diff
@@ -126,3 +126,58 @@ function gutenberg_legacy_widget_settings( $settings ) {
 	return $settings;
 }
 add_filter( 'block_editor_settings', 'gutenberg_legacy_widget_settings' );
+
+
+function create_wp_area_post_type() {
+	register_post_type(
+		'wp_area',
+		array(
+			'labels'                => array(
+				'name'                     => _x( 'WordPress Area', 'post type general name' ),
+				'singular_name'            => _x( 'WordPress Area', 'post type singular name' ),
+				'menu_name'                => _x( 'WordPress Areas', 'admin menu' ),
+				'name_admin_bar'           => _x( 'WordPress Area', 'add new on admin bar' ),
+				'add_new'                  => _x( 'Add New', 'Block' ),
+				'add_new_item'             => __( 'Add New WordPress Area' ),
+				'new_item'                 => __( 'New WordPress Area' ),
+				'edit_item'                => __( 'Edit WordPress Area' ),
+				'view_item'                => __( 'View WordPress Area' ),
+				'all_items'                => __( 'All WordPress Areas' ),
+				'search_items'             => __( 'Search WordPress Areas' ),
+				'not_found'                => __( 'No WordPress area found.' ),
+				'not_found_in_trash'       => __( 'No WordPress areas found in Trash.' ),
+				'filter_items_list'        => __( 'Filter WordPress areas list' ),
+				'items_list_navigation'    => __( 'WordPress areas list navigation' ),
+				'items_list'               => __( 'WordPress areas list' ),
+				'item_published'           => __( 'WordPress area published.' ),
+				'item_published_privately' => __( 'WordPress area published privately.' ),
+				'item_reverted_to_draft'   => __( 'WordPress area reverted to draft.' ),
+				'item_scheduled'           => __( 'WordPress area scheduled.' ),
+				'item_updated'             => __( 'WordPress area updated.' ),
+			),
+			'public'                => false,
+			'show_ui'               => true,
+			'show_in_menu'          => false,
+			'show_in_rest'          => true,
```

Comment:
Are we following-up with the REST endpoints PR? we might want to disable the built-in ones if it's the case?

Reply:
My plan was to leave the built-in ones. The REST endpoints that we are following up are specific to widget areas, and they receive as the identifier a widget area name. What if I want to change wp_area by its id (numeric)? What if I want to do other actions (list all areas widget/navigations..., search areas or use other nice features from the posts endpoint)?
It seemed to allow direct access to the wp_areas endpoint did have big cons and had some advantages.
The new endpoint may end up referencing the same thing but it references using a different identifier, and the new endpoint is widget specific (returning widget specific information besides the post itself being referenced)